src/main/scala/org/broadinstitute/dsde/workbench/avram/api/AvramServlet.scala

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.dsde.workbench.avram.api
+
+import java.util.logging.Logger
+
+import scala.concurrent.duration.Duration
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import org.broadinstitute.dsde.workbench.avram.Avram
+import org.broadinstitute.dsde.workbench.avram.dataaccess.SamUserInfoResponse
+import org.broadinstitute.dsde.workbench.avram.model.AvramException
+
+import scala.concurrent.Await
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+
+trait AvramServlet {
+
+  implicit val executionContext = scala.concurrent.ExecutionContext.Implicits.global
+
+  val log = Logger.getLogger(getClass.getName)
+  private val samDao = Avram.samDao
+  private val bearerPattern = """(?i)bearer (.*)""".r
+
+  def registerOnComplete[T](future: Future[T], request: HttpServletRequest, response: HttpServletResponse) = {
+    future onComplete {
+      case Success(result) => {
+        response.setStatus(HttpServletResponse.SC_OK)
+      }
+      case Failure(e: AvramException) => {
+        response.setStatus(e.status)
+        response.getWriter.write(e.message)
+      }
+      case Failure(e) => {
+        response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR)
+        response.getWriter.write(e.getMessage)
+      }
+    }
+    future
+  }
+
+  def handleAuthenticatedRequest[T]
```

Comment:
I'm actually not certain the structure between `handleAuthenticatedRequest` and `registerOnComplete` is correct. If we fail to resolve a token with Sam, then the `unsafeRun` here will throw an exception. I don't think this exception is caught, so would result in a 500 error to the user.

Reply:
Yeah, that's a good observation. I _think_ this will be sufficiently different in an _IO_ world that we're better off addressing it then, but I'm not 100% sure.

Reply:
OK, we can table this to a post-IO world.

Reply:
What I'd love to see in a post-IO world is a structure like this (psuedocode):
```
  override def doGet(request: HttpServletRequest, response: HttpServletResponse) = {
    val json = for {
      userInfo <- authWithSam(request)
      externalId <- parseExternalIdParam(request)
      dbCollection <- collectionsService.getCollection(externalId)
    } yield dbCollection.asJson.noSpaces
    
    commitResponse(json, request, response)  // the "unsafe" part
  }
```